cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -539,33 +539,31 @@ Status NumPyConverter::Visit(const BinaryType& type) {
 
   auto data = reinterpret_cast<const uint8_t*>(PyArray_DATA(arr_));
 
-  int item_length = 0;
+  auto AppendNotNull = [&builder, this](const uint8_t* data) {
+    // This is annoying. NumPy allows strings to have nul-terminators, so
+    // we must check for them here
+    int item_length = 0;
```

Comment:
Could perhaps simply use the [`strnlen_s`](https://en.cppreference.com/w/c/string/byte/strlen) function.

Reply:
is it ok to use `strnlen`? `strnlen_s` seems to be C11 only (though does C++11 imply that C11 extensions are available)?

Reply:
Oh... I had read "C++11". Let's forget about it :-)